src/centralstore/incremental/NodeManager.cpp

```diff
@@ -0,0 +1,254 @@
+/**
+Copyright 2020 JasmineGraph Team
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+**/
+
+#include "NodeManager.h"
+
+#include <sys/stat.h>
+
+#include "../../util/logger/Logger.h"
+#include "NodeBlock.h"  // To setup node DB
+#include "PropertyLink.h"
+#include "RelationBlock.h"
+
+Logger node_manager_logger;
+
+NodeManager::NodeManager(std::string mode) {
+    std::ios_base::openmode openMode = std::ios::trunc;  // default is Trunc mode which overrides the entire file
+    if (mode == "app") {
```

Comment:
Shall we move the literal `"app"` to a constant inside the `NodeManager.h` file?

Reply:
Yes, We can move this to a static constant, But in header, only static variable definition is allowed. ~[source](https://stackoverflow.com/questions/1563897/static-constant-string-class-member)

Reply:
Fixed in https://github.com/miyurud/jasminegraph/pull/115/commits/ed31ba0af33e9629173808960eef3cfe90c5dbe9

Reply:
Done